matchms/filtering/peak_processing/require_number_of_peaks_below_maximum.py

```diff
@@ -0,0 +1,29 @@
+import logging
+from typing import Optional
+from matchms.Spectrum import Spectrum
+
+
+logger = logging.getLogger("matchms")
+
+
+def require_number_of_peaks_below_maximum(spectrum_in: Spectrum,
```

Comment:
```suggestion
def require_upper_limit_of_number_of_peaks(spectrum_in: Spectrum,
```

Reply:
Also not 100% happy with my suggestion. But I first misunderstood your function name and thought you meant only peaks below the max intensity or something like this...

Reply:
@florian-huber How about require_less_than_maximum_number_of_peaks?